Dev sort by workshop area

jjacobs10 opened the pull request:
[Wiki page](https://github.com/cseseniordesign/reservations/wiki/Story-%2365-Tabs-for-sorting-by-workshop-area)

jjacobs10 committed:
Attempt

Attempted to copy the "Studio Status" box from the Users page. Tried several things; deleted most of the dead code before pushing.

adobrusky committed:
Fixed issue with workshop category dropdown not working

jjacobs10 committed:
Made dropdown dynamic; fixed filter bug

Previously once a filter was applied a user could see tools they were not authorized to use.

jjacobs10 committed:
More clean up

jjacobs10 committed:
Filters for Events and Tools page

jjacobs10 committed:
Clean up

adobrusky commented on views/tools.erb:
```diff
@@ -1,5 +1,28 @@
+                    </select>
+
+                    <%# Assign the selected category to the `workshop_category` variable. %>
+                    <input type="hidden" name="hidden-workshop-category" id="hidden-workshop-category" value="<%= workshop_category %>">
```
You don't need this hidden field. Hidden fields are typically just used to get values that are passed from the .rb into javascript. So like if you wanted to use that workshop_category variable value inside of some javascript code you would use a hidden field.

adobrusky commented on views/admin/events.erb:
```diff
@@ -32,7 +32,30 @@
+                                        </select>
+
+                                        <%# Assign the selected type to the `event_type` variable. %>
+                                        <input type="hidden" name="hidden-event-type" id="hidden-event-type" value="<%= event_type %>">
```
You don't need this hidden field. Hidden fields are typically just used to get values that are passed from the .rb into javascript. So like if you wanted to use that event_type variable value inside of some javascript code you would use a hidden field.

jjacobs10 replied:
Sponsors decided they do not want the Event Type dropdown so all of the related code has been removed. So this is resolved.

adobrusky reviewed (commented):
Overall code looks good besides a couple small things

jjacobs10 committed:
Fixed several issues

Removed unnecessary hidden fields.

Fixed the Event Type filtering not working for past events and when navigating using pagination.

Removed the Event Type dropdown from the Agenda page as per the sponsors' request.

Fixed an issue with the Agenda page's date and workshop filter not working at the same time.

jjacobs10 committed:
Fixed workshop category filter not working on Agenda page

Fixed workshop category filter not working on the Agenda page.

Removed code for filtering events on the Agenda page.

Added a line to the Tools page's workshop category definition to match the earlier definition of the tools object.

jjacobs10 committed:
Added comment

adobrusky commented on views/admin/agenda.erb:
```diff
@@ -1,9 +1,26 @@
+                </select>
+
+                <%# Assign the selected category to the `workshop_category` variable. %>
+                <input type="hidden" name="hidden-workshop-category" id="hidden-workshop-category" value="<%= params[:workshop_category] %>">
```
Can remove this hidden field since it is unused

adobrusky commented on routes/admin/events.rb:
```diff
@@ -36,13 +37,19 @@
+
+	# Redefine the iterator variable if there is an event type filter applied
+	unless event_type.nil? || event_type.length == 0
+		iterator = Event.includes(:event_signups).where(:service_space_id => SS_ID).where(where_clause).where(:event_type_id => event_type)
```
Minor but could simplify this to ```iterator = iterator.where(:event_type_id => event_type)```

adobrusky commented on views/admin/events.erb:
```diff
@@ -31,8 +31,33 @@
+                                            <% end %>
+                                        </select>
+
+                                        <%# Assign the selected type to the `event_type` variable. %>
```
I think this comment is here because before the hidden field was for storing event_type but we changed it to tab so now it no longer applies

adobrusky commented on views/admin/agenda.erb:
```diff
@@ -1,9 +1,26 @@
+            <div style="width:200px">
+                <label for="workshop-category">Workshop Category:</label>
+                <select id="workshop-category" tabindex="1" name="workshop_category" class="dcf-input-select">
+                    <option value=""></option>
```
Should this be "All" instead of being blank?

adobrusky commented on views/admin/events.erb:
```diff
@@ -31,8 +31,33 @@
+                                    <label for="event-type">Event Type:</label>
+                                    <div class="dropdown-container" style="display:flex;">
+                                        <select id="event-type" style="max-width:200px;min-width:50px;margin-top:10px;" tabindex="1" name="event_type" class="dcf-input-select">
+                                            <option value=""></option>
```
Should this be "All" instead of being blank?

adobrusky commented on views/tools.erb:
```diff
@@ -1,5 +1,24 @@
+                <div>
+                    <label for="workshop-category">Workshop Category:</label>
+                    <select id="workshop-category" style="width:auto;margin-top:10px;" tabindex="1" name="workshop_category" class="dcf-input-select">
+                    <option value=""></option>
```
Should this be "All" instead of being blank?

adobrusky reviewed (approved):
Excellent work overall just a couple of small things then it should be good to merge.

jjacobs10 committed:
Made changes according to reviewer's comments

- Changed default dropdown option to be "All" instead of blank

- Simplified the `iterator` variable declaration

adobrusky committed:
Merge branch 'stable' into dev-sort-by-workshop-area
